x/marketmap/types/market.go

```diff
@@ -40,6 +40,48 @@ func (mm *MarketMap) ValidateBasic() error {
 	return nil
 }
 
+// GetValidSubset outputs a MarketMap which contains the maximal valid subset of this MarketMap.
+//
+//	In particular, this will eliminate anything which would otherwise cause a failure in ValidateBasic.
+//	The resulting MarketMap should be able to pass ValidateBasic.
+func (mm *MarketMap) GetValidSubset() MarketMap {
```

Comment:
I feel like this function should return an `error` as well and call `Validate` to ensure that the returned map is valid

Reply:
There is a case where there is no valid subset and we should just return `nil, error`

Reply:
I think you're right--if the entire set of markets is invalid we should just keep everything the same. Returning an error will do that while I think returning an empty subset would just stop fetching any prices at all.